evalml/pipelines/components/estimators/regressors/arima_regressor.py

```diff
@@ -129,6 +132,23 @@ def _set_forecast(self, X):
         fh_ = ForecastingHorizon([i + 1 for i in range(len(X))], is_relative=True)
         return fh_
 
+    def _get_sp(self, X):
+        if X is None:
+            return 1
+        freq_mappings = {
+            "D": 7,
+            "W": 52,
+            "M": 12,
+            "Q": 4,
+        }
+        time_index = self._parameters.get("time_index", None)
+        sp = self.arima_parameters["sp"]
+        if sp == "detect":
+            inferred_freqs = X.ww.infer_temporal_frequencies()
+            freq = inferred_freqs.get(time_index, None)
+            sp = freq_mappings.get(freq, 1)
```

Comment:
this handles the case where the freq is `None` right? Just for my own clarity 😄

Reply:
Yep, thanks to the `get`, a frequency of None will default to 1 (no seasonality)!